### PR TITLE
Split out Todoist samples and reorganize

### PR DESCRIPTION
**Pull request by ekoleda-codaio**
In #1460 I added a "full" Todoist sample. This creates smaller, topical samples based off of that code, and then integrates them into the corresponding topic pages. I also removed some topical samples that were duplicative of the snippets, and just used that snippet instead.

Staged: https://coda-packs-sdk.readthedocs-hosted.com/en/ek-todoistsplit/samples/topic/authentication/

### CODE REVIEW

File: docs/samples/topic/sync-table.md [hunk 1]
```diff
@@ -2,69 +2,126 @@
...
-      // recreate them from scratch.
-      id: "<idColumn>",
-      // The property name from the properties object below that should label this
-      // item in the UI. All properties can be seen when hovering over a synced
-      // item in the UI, but the primary property value is shown on the chip
-      // representing the full object.
-      primary: "<displayColumn>",
-    })
+      id: "property1", // Which property above is a unique ID.
+      primary: "property2", // Which property above to display by default.
```
Comment (huayang-codaio) [inline docs/samples/topic/sync-table.md:15]: better!

File: docs/samples/topic/dynamic-sync-table.md [hunk 2]
```diff
@@ -12,89 +12,67 @@ There are two subtle variants of dynamic sync tables. A sync table can be dynami
...
-            // this item in the UI. All properties can be seen when hovering over a
-            // synced item in the UI, but the primary property value is shown on the
-            // chip representing the full object.
-            primary: "<displayColumn>",
-            // The columns that will be displayed on the table
-            featured: ["<otherColumn>"],
+            properties: propreties,
+            id: id,
+            primary: primary,
+            featured: featured,
```
Comment (huayang-codaio) [inline docs/samples/topic/dynamic-sync-table.md:46]: guess this is not completed yet but usually it's like `{properties, id, primary, featured}`

Comment (ekoleda-codaio) [reply docs/samples/topic/dynamic-sync-table.md:46]: This is intentional, as per the go/sample-code-style-guide. While the shorthand notation is nice, I saw a lot of cases in the Packathon where it led to confusion.

File: docs/samples/topic/column-format.md [hunk 3]
```diff
@@ -4,18 +4,119 @@ A **column format** is a custom column type that you apply to any column in any
...
-    // A custom column type that you apply to any column in any Coda table.
     pack.addColumnFormat({
-      name: "<format name>",
-      instructions: "<instruction text that will be visible by Users>",
-      formulaName: "<the name of a formula to execute using the column value>",
-      formulaNamespace: "Ignore", // this will be deprecated
-      // Optional, regular expressions that detect urls that are handeable by this
-      // column format. When a user pastes a url that matches any of these, the UI
-      // will hint to the user that they can apply this format to their column.
-      matchers: [],
```
[the remark this answers is not in the document]
Comment (ekoleda-codaio) [reply docs/samples/topic/column-format.md:19]: It won't match any URLs. At the moment the matchers are only used for guessing the column type based on the first data entered, and the Pack must already be installed.

File: docs/samples/topic/sync-table.md (hunk 1 above)
Comment (ekoleda-codaio) [reply docs/samples/topic/sync-table.md:15]: 😄

**Pull request merged**